components/pedigree/resources/src/main/resources/pedigree/view/cancersLegend.js

```diff
@@ -60,63 +83,112 @@ define([
 
         _getAllSupportedCancers: function() {
             var clist = [];
-            for (var cancer in this._cancerColors) {
-                if (this._cancerColors.hasOwnProperty(cancer)) {
+            for (var cancer in this._customCancerColors) {
+                if (this._customCancerColors.hasOwnProperty(cancer)) {
                     clist.push(cancer);
                 }
             }
             return clist;
         },
 
+        _isSupportedCancer: function(cancerId) {
+            return this._customCancerColors.hasOwnProperty(cancerId);
+        },
+
+        /**
+         * Returns the HPOTerm object for the cancer with the given ID. If object is not in cache yet
+         * returns a newly created one which may have the cancer name & other attributes not loaded yet
+         *
+         * @method getCancer
+         * @return {Object}
+         */
+        getCancer: function(cancerID) {
+            if (!this._termCache.hasOwnProperty(cancerID)) {
+                var whenNameIsLoaded = function() { this._updateCancerName(cancerId); };
+                var name = this._customLabelMapping.hasOwnProperty(cancerID) ? this._customLabelMapping[cancerID] : null;
+                this._termCache[cancerID] = new HPOTerm(cancerID, name, whenNameIsLoaded.bind(this));
+            }
+            return this._termCache[cancerID];
+        },
+
         /**
          * Returns cancer name
          *
-         * @param {Number|String} ID for this cancer
+         * @param {Number|String} cancerID for this cancer
          * @return {String} associated cancer name
          */
         getName: function($super, cancerID) {
-            return this._cancerLabels.hasOwnProperty(cancerID)? this._cancerLabels[cancerID] : cancerID;
+            var stored = this.getCancer(cancerID).getName();
+            return this._customCancerLabels.hasOwnProperty(stored) ? this._customCancerLabels[stored] : stored;
         },
 
+        /**
+         * Registers an occurrence of a cancer. If disorder hasn't been documented yet,
+         * designates a color for it.
+         *
+         * @method addCase
+         * @param {Number|String} id ID for this cancer taken from the HPO database
+         * @param {String} name The name of the cancer
+         * @param {Number} nodeID ID of the Person who has this disorder
+         */
         addCase: function($super, id, name, nodeID) {
-            var name = this.getName(name);
+            if (!this._termCache.hasOwnProperty(id))
+                this._termCache[id] = new HPOTerm(id, name);
```

Comment:
I think this is the same code as in HPOLegend, maybe this legend could have extended HPOLegend to avoid this duplication of HPO loading. Not sure it would work nice, will take a look

Reply:
Just a note, the final version of this may end up not using HPO terms. From what I understood, there are several ontologies that are currently under consideration (HPO, NCI, OncoTree).